blog/_posts/2021-10-30-debuging-root-scripts-in-eclipse.md

```diff
@@ -0,0 +1,227 @@
+
+<center>
+{% include figure_image
+   img="/assets/images/eclipse-02-stop-indexer.png"
+   style="width: 100%"
+   caption="Stop indexing operation. The indexer will restart after the Debug configuration is created."
+%}
+</center>
+
+### Setting up Debug and Run Configurations
+
+We will start by setting up the main Debug configuration for Geant4 `root-eclipse` program.
+
+<center>
+{% include figure_image
+   img="/assets/images/eclipse-03-build.png"
+   style="width: 100%"
+   caption="Expand 'Build Targets' and invoke 'all' target. Executable will be compiled."
+%}
+</center>
+
+1. Expand 🎯 Build Targets under the main project node in the Project Explorer window. Double-click "All" to compile the executable.
+2. In the menu "Run" open "Debug configurations". Right-click "C/C++ Application" and select "New Configuration".
+3. Click "Browse" next to the "Project" field. Select "root-eclipse-Debug@eclipse" in the modal dialog.
+4. Click "Search Project" next to the "C/C++ Application" field. Specify the "root-eclipse" executable built earlier.
+
+<center>
+{% include figure_image
+   img="/assets/images/eclipse-04-debug-configuration.png"
+   style="width: 100%"
+   caption="Set up a new Debug configuration."
+%}
+</center>
+
+Finally, we can run the project in Debug mode. In Eclipse menu select `Run → Debug`. Eclipse will run the project and simultaneously start indexing all ROOT source files. Depending on the speed of your hard drive and memory indexing will require from several minutes to about one hour.
+
+### Debugging Your Personal Script
+
+If you originally have a standalone ROOT-based C++ project with CMake, then everything should work out of the box and you can [skip to the next section](https://root.cern/blog/debuging-root-scripts-in-eclipse#ramdisk-for-older-computers).
+
+Next we will discuss an alternative option. You don't have a CMake based project and want to use my Template GitHub project. Then your ROOT script needs to be properly integrated into the existing project. Generally speaking, this involves following:
+
+* Specify extra libraries you are utilizing in project's `CMakeLists.txt`.
+* Explicitly defne all the headers used in your script (Cling does not require that).
+* Indicate certain class names in project's `LinkDef.h` file for the proper dictionary and shared library generation.
+
+A detailed instrutions elaborating each item above can be found in my Template GitHub Project's [readme file](https://github.com/petrstepanov/root-eclipse#debugging-your-personal-script).
+
+Now we are ready to debug your ROOT script. Save changes in all modified source files. In Eclipse menu select Project > Build All (or run the `all` target under the Build Targets in the "Project Explorer" window). Finally, to start debugging run the previously created debug configuration in Run > Debug menu item.
+
+## RAMDISK for Older Computers
```

Comment:
Is there a way to exclude `interpreter/llvm` from indexing?

Reply:
Thank you, I updated the Makefile to exclude the `interpreter/llvm` folder.